lib/datum.js

```diff
@@ -18,23 +18,27 @@ var datum = function(proj) {
   if (proj.datumCode && proj.datumCode === 'none') {
     this.datum_type = PJD_NODATUM;
   }
+
   if (proj.datum_params) {
-    for (var i = 0; i < proj.datum_params.length; i++) {
-      proj.datum_params[i] = parseFloat(proj.datum_params[i]);
+    // Copy datum_params object to fix issue with custom 7 point projection (https://github.com/proj4js/proj4js/issues/141)
```

Comment:
we don't need this comment, comments like these have a tendency to pile up over time to make the code hard to read